tools/esdlcmd/esdlcmd_manifest.cpp

```diff
@@ -0,0 +1,1409 @@
+                                    appendEndTag(etag);
+                                level--;
+                                break;
+
+                            case XmlPullParser::END_DOCUMENT:
+                                level = 0;
+                                break;
+                        }
+
+                        lastType = type;
+
+                        #ifdef _MANIFEST_DEBUG_
+                        UERRLOG("processBinding while:(...%s)", binding.length()<100 ? binding.str() : binding.str()+binding.length()-100);
+                        #endif
+                    }
+                    while (level > 0);
+
+                    if (isBundle)
+                        appendEndTag(MANIFEST_TAG_BUNDLE);
+                    if (isBundle && !foundServiceDefn)
+                        UWARNLOG("Generating EsdlBundle output without ESDL Service Definition. This may cause runtime issues.");
+
+                    #ifdef _MANIFEST_DEBUG_
+                    UERRLOG("processBinding final:(%s)", binding.str());
+                    #endif
+                }
+                else
+                {
+                    UERRLOG("Error reading manifest file (%s): empty or does not exist.", optManifestPath.str());
+                    result = false;
+                }
+            }
+            catch (IException* e)
+            {
+                StringBuffer msg;
+                e->errorMessage(msg);
+                UERRLOG("Error reading manifest file (%s): %s", optManifestPath.str(), msg.str());
+                result = false;
+                e->Release();
+            }
+            catch (XmlPullParserException& xe)
+            {
+                const char* msg = xe.what();
+                UERRLOG("Error reading manifest file (%s) at line #%d, column #%d: %s", optManifestPath.str(), xe.getLineNumber(), xe.getColumnNumber(), msg ? msg : "Unknown");
+            }
+            catch (...)
+            {
+                UERRLOG("Unknown error reading manifest file (%s)", optManifestPath.str());
+                result = false;
+            }
+
+            return result;
+        }
+
+        bool outputResult()
+        {
+            bool result = true;
+            try {
+                if (optOutputPath.isEmpty())
+                {
+                    printf("%s", binding.str());
+                }
+                else
+                {
+                    Owned<IFile> bindingFile = createIFile(optOutputPath.str());
+                    if (bindingFile)
+                    {
+                        Owned<IFileIO> bindingFileIO = bindingFile->open(IFOcreate);
+                        if (bindingFileIO)
+                        {
+                            bindingFileIO->write(0, binding.length(), binding.str());
+                        } else {
+                            UERRLOG("Error writing to binding file (%s).", optOutputPath.str());
```

Comment:
Ignoring the potential of other artifact types, not all output files are binding files. Better to keep it generic and omit "binding".